web/packages/teleport/src/AuthConnectors/AuthConnectors.tsx

```diff
@@ -26,73 +30,106 @@ import {
 import { FeatureBox, FeatureHeaderTitle } from 'teleport/components/Layout';
-import ResourceEditor from 'teleport/components/ResourceEditor';
+import { Route, Switch } from 'teleport/components/Router';
 import useResources from 'teleport/components/useResources';
+import cfg from 'teleport/config';
+import { Resource } from 'teleport/services/resources';
+import useTeleport from 'teleport/useTeleport';
 
+import { GitHubConnectorEditor } from './AuthConnectorEditor';
 import ConnectorList from './ConnectorList';
-import CTAConnectors from './ConnectorList/CTAConnectors';
+import { CtaConnectors } from './ConnectorList/CTAConnectors';
 import DeleteConnectorDialog from './DeleteConnectorDialog';
 import EmptyList from './EmptyList';
 import templates from './templates';
-import useAuthConnectors, { State } from './useAuthConnectors';
+
+export const description =
+  'Auth connectors allow Teleport to authenticate users via an external identity source such as Okta, Microsoft Entra ID, GitHub, etc. This authentication method is commonly known as single sign-on (SSO).';
 
 export function AuthConnectorsContainer() {
-  const state = useAuthConnectors();
-  return <AuthConnectors {...state} />;
+  return (
+    <Switch>
+      <Route
+        key="auth-connector-edit"
+        path={cfg.routes.ssoConnector.edit}
+        render={() => <GitHubConnectorEditor />}
+      />
+      <Route
+        key="auth-connector-new"
+        path={cfg.routes.ssoConnector.create}
+        render={() => <GitHubConnectorEditor isNew={true} />}
+      />
+      <Route
+        key="auth-connector-list"
+        path={cfg.routes.sso}
+        exact
+        render={() => <AuthConnectors />}
+      />
+    </Switch>
+  );
 }
 
-export function AuthConnectors(props: State) {
-  const { attempt, items, remove, save } = props;
-  const isEmpty = items.length === 0;
-  const resources = useResources(items, templates);
+export function AuthConnectors() {
+  const ctx = useTeleport();
+  const [items, setItems] = useState<Resource<'github'>[]>([]);
 
-  const title =
-    resources.status === 'creating'
-      ? 'Creating a new github connector'
-      : 'Editing github connector';
-  const description =
-    'Auth connectors allow Teleport to authenticate users via an external identity source such as Okta, Microsoft Entra ID, GitHub, etc. This authentication method is commonly known as single sign-on (SSO).';
+  const [fetchAttempt, fetchConnectors] = useAsync(async () => {
+    const response = await ctx.resourceService.fetchGithubConnectors();
+    setItems(response);
+  });
 
-  function handleOnSave(content: string) {
-    const name = resources.item.name;
-    const isNew = resources.status === 'creating';
-    return save(name, content, isNew);
+  function remove(name: string) {
+    return ctx.resourceService
+      .deleteGithubConnector(name)
+      .then(fetchConnectors);
   }
 
+  useEffect(() => {
+    fetchConnectors();
+  }, []);
```

Comment:
nit: Only invoke `fetchConnectors` if `fetchAttempt.status !== sucess`?

Reply:
hmm I'm not sure what purpose this would serve, since the `fetchAttempt` will never be `success` on first render. This isn't a pattern i've seen anywhere else in the codebase